### ISSUE TITLE
[php] Socket.select(), Socket.setFastSend() and Socket.listen() are not implemented

### ISSUE THREAD

**Issue by RealyUniqueName** [HaxeFoundation/haxe#6816]

**Comment by ncannasse** [comment]
I did open the link with the hope it was a pull request :)

**Comment by RealyUniqueName** [comment]
Are you using php target? )

**Comment by ncannasse** [comment]
No, but it would have been a simple merge ;)

**Comment by bendmorris** [comment]
@RealyUniqueName are we open to requiring the sockets extension for the PHP target? It's compiled in by default and my basic Ubuntu install has it enabled, but some do disable it.

If so this will be trivial to implement, but if not I think getting `bind` and `listen` to work consistent with other targets will be impossible, and it looks like TCP_NODELAY support wasn't added for streams until 7.1 (https://bugs.php.net/bug.php?id=51879)

**Comment by RealyUniqueName** [comment]
Yes, socket extension is ok. We already require mysql, sqlite and multibyte string extensions. And `stream_socket_*` functions also belong to an extension.

**Comment by bendmorris** [comment]
Addressing as part of https://github.com/HaxeFoundation/haxe/pull/6887
